Control coverage/Feature/NSG.md

```diff
@@ -0,0 +1,7 @@
+## NSG
+
+| ControlId | Dependent Azure API(s) and Properties | Control spec |
+|-----------|-------------------------------------|------------------|
+| <b>ControlId:</b><br>Azure_NSG_NetSec_Dont_Open_InBound_Any_Any<br><br><b>DisplayName:</b><br>Firewall/NSG rules must not allow unrestricted traffic (any-any rule).<br><br><b>Description: </b><br> Do not leave restricted ports open on NSG for inbound traffic. |<b>ARM API to lists all the NSG configurations under<br> the specified subscription:</b> <br> /subscriptions/{subscriptionId}/providers<br>/Microsoft.Network/networkSecurityGroups<br>?api-version=2019-04-01 <br><br><b>Properties:</b><br> properties.destinationPortRanges| <b>Scope: </b> All NSGs in subscription.<br><br> <b>Config: </b> ExclusionTags: <br> 1. Description: VM is part of ADB cluster. <br> TagName: databricks-environment <br> TagValue: true <br> 2. Description: VM is part of ADB cluster. <br> TagName: application <br> TagValue: databricks <br><br><b>Passed: </b><br>1. No inbound rule defined. <br> *or* <br> 2. Any-any inbound rule not found.<br><br><b>Failed: </b><br>Any-any inbound rule found.<br><br><b>Not  applicable: </b><br> Network Security Groups(NSG) associated with Virtual Machine is part of Azure Databricks (ADB) cluster. <br><br>*Note: It won't check for the rule's priority. It considers all the active inbound rules configured.*|
+| <b>ControlId:</b><br>Azure_NSG_NetSec_Dont_Open_Restricted_Ports<br><br><b>DisplayName:</b><br>Do not use risky ports on firewall and NSGs.<br><br><b>Description: </b><br> Do not leave restricted ports open on NSG. |<b> ARM API to lists all the NSG configurations under<br> the specified subscription:</b> <br> /subscriptions/{subscriptionId}/providers<br>/Microsoft.Network/networkSecurityGroups<br>?api-version=2019-04-01 <br><br><b>Properties:</b><br> properties.destinationPortRanges | <b>Scope: </b> All NSGs in subscription.<br><br> <b>Config: </b> RestrictedPorts: 445, 3389, 5985, 22<br><br><b>Passed: </b><br>1. If no restricted port [445,3389,5985,22] found. <br> *or* <br> 2. If Any-Any inbound rule not found.<br><br><b>Failed: </b><br>1. Any-any inbound rule found. <br> *or* <br> 2. Any restricted port found in the inbound security rule. <br><br>*Note: It won't check for the rule's priority. It considers all the active inbound rules configured. This does not consider default inbound rule.*|
```

Comment:
For ports we check for both properties destinationPortRanges & destinationPortRange.

Reply:
Added one more property 'destinationPortRange'.